lang/py/src/avro/schema.py

```diff
@@ -405,6 +419,102 @@ def __eq__(self, that):
     return self.props == that.props
 
 #
+# Decimal Type
+#
+class DecimalSchema(LogicalSchema):
+  def __init__(self, decimal_type, precision, scale):
+    if decimal_type != 'fixed' and decimal_type != 'bytes':
+      raise SchemaParseException('Logical type DECIMAL must be backed by fixed or bytes.')
+
+    LogicalSchema.__init__(self, decimal_type, 'decimal')
+    self.set_prop('precision', precision)
+    self.set_prop('scale', scale)
+
+  def validate(self, validation_props=None):
```

Comment:
another way to define this function signature is `def validate(self, **props_to_validate)`.  (not a blocker)